OloEngine/src/OloEngine/Scene/Scene.cpp

```
@@ -344,7 +346,32 @@ namespace OloEngine
         if (!entity || !entity.HasComponent<IDComponent>())
             return;
 
+        // Dispatch Lua OnDestroy before the entity is removed from the registry
+        if (m_IsRunning && entity.HasComponent<LuaScriptComponent>())
+        {
+            if (auto const& lsc = entity.GetComponent<LuaScriptComponent>(); !lsc.ScriptFile.empty())
+            {
+                LuaScriptEngine::OnDestroyEntity(entity);
```

Clang (14.0.6)

[warning] variable 'entity' is not initialized

(cppcoreguidelines-init-variables)